source/simulator/assembly.cc

```diff
@@ -1521,7 +1521,7 @@ namespace aspect
       }
     else
       {
-        compute_material_model_input_values (old_solution,
+        compute_material_model_input_values (current_linearization_point,
```

Comment:
You can remove this from the if-loop now, and only keep things in there that are specific to the temperature system. By the way, are the explicit_material_model terms specific to the temperature system? Or do we need an explicit material model for the composition system now as well (I think we use it for the stabilization calculation).